templates/test/prow-machinepool-ci-version/patches/ci-version.yaml

```diff
@@ -0,0 +1,241 @@
+apiVersion: controlplane.cluster.x-k8s.io/v1alpha3
+kind: KubeadmControlPlane
```

Comment:
Is there a way to reuse the patch for KubeadmControlPlane and KubeadmConfig between machine pool and non-machine pool templates? If the only thing that changes in here from `templates/test/prow-ci-version/patches/ci-version.yaml` is the `AzureMachinePool` we can put that in its own patch file and list the two patch files as patches in kustomization.yaml.

Reply:
I think patches for `KubeadmControlPlane` and control plane's `AzureMachineTemplate` can be shared between machine pool and non-machine pool template. 

For machine pool template, it uses `KubeadmConfig` for bootstraping machinepool, whereas non-machinepool template uses `KubeadmConfigTemplate` to bootstrap machine deployment. Not sure what the difference between them is so I am not sure if we can re-use them :/

Reply:
@devigned @juan-lee I need your help refreshing my memory here, do you know why MachinePools don't use `KubeadmConfigTemplate` as the bootstrap object ref type in  `MachineSpec`?

Reply:
I was curious about the same thing. It actually threw me off during the implementation.

Reply:
For the record, I just tried changing MachinePools to use KubeadmConfigTemplate and it didn't work, Bootstrap data never became available. Didn't dig deeper than that though. Would be good to understand the difference.